hudi-common/src/main/java/org/apache/hudi/io/storage/HoodieAvroParquetReader.java

```diff
@@ -165,7 +165,10 @@ private ClosableIterator<IndexedRecord> getIndexedRecordIteratorInternal(Schema
       AvroReadSupport.setAvroReadSchema(conf, requestedSchema.get());
       AvroReadSupport.setRequestedProjection(conf, requestedSchema.get());
     }
-    ParquetReader<IndexedRecord> reader = new HoodieAvroParquetReaderBuilder<IndexedRecord>(path).withConf(conf).build();
+    ParquetReader<IndexedRecord> reader = new HoodieAvroParquetReaderBuilder<IndexedRecord>(path).withConf(conf)
+        .set(ParquetInputFormat.STRICT_TYPE_CHECKING, "false")
+        .set(AvroSchemaConverter.ADD_LIST_ELEMENT_RECORDS, "false")
```

Comment:
Why do we need to set this? Also, why isn't this guarded by schema evol flag? Do we want to hardcode these configs in all cases?

Reply:
tryOverrideDefaultConfigs pretty much hardcodes this. But tryOverrideDefaultConfigs is wrong. It is not setting the right config. I stepped through the code with the debugger and this is how you set it correctly. This is no longer relevant for this pr since I am focused on row writer enable so maybe I should revert this and make a jira?

Reply:
Going to add a jira and remove from this pr